pkg/controller/batch/computeenvironment/setup.go

```diff
@@ -0,0 +1,553 @@
+			e.postDelete = postDelete
+		},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(o.ForControllerRuntime()).
+		For(&svcapitypes.ComputeEnvironment{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.ComputeEnvironmentGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithInitializers(
+				managed.NewDefaultProviderConfig(mgr.GetClient()),
+				managed.NewNameAsExternalName(mgr.GetClient())),
+			managed.WithPollInterval(o.PollInterval),
+			managed.WithLogger(o.Logger.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.ComputeEnvironment, obj *svcsdk.DescribeComputeEnvironmentsInput) error {
+	obj.ComputeEnvironments = []*string{awsclients.String(meta.GetExternalName(cr))} // we only want to observe our CE
+	fmt.Printf("preObserve: obj.ComputeEnvironments[0]: %v\n", awsclients.StringValue(obj.ComputeEnvironments[0]))
+	fmt.Println("preObserve: cr.Spec.ForProvider.ServiceRoleARN ", awsclients.StringValue(cr.Spec.ForProvider.ServiceRoleARN))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.ComputeEnvironment, resp *svcsdk.DescribeComputeEnvironmentsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	fmt.Println("postObserve: meta.GetExternalName(cr)", meta.GetExternalName(cr))
+	fmt.Printf("postObserve start: obs: exists?, uptodate?, lateinitialized? : %v\n", obs)
+
+	fmt.Printf("postObserve: resp describeout: %v\n", resp)
+
+	fmt.Println("postObserve: current resp.ComputeEnvironments[0].State ", awsclients.StringValue(resp.ComputeEnvironments[0].State))
+	fmt.Println("postObserve: current cr.Status.AtProvider.State ", awsclients.StringValue(cr.Status.AtProvider.State))
+	fmt.Println("postObserve: cr.Status.AtProvider.ComputeEnvironmentARN ", awsclients.StringValue(cr.Status.AtProvider.ComputeEnvironmentARN))
+	fmt.Println("postObserve: cr.Spec.ForProvider.ServiceRoleARN ", awsclients.StringValue(cr.Spec.ForProvider.ServiceRoleARN))
+
+	switch awsclients.StringValue(resp.ComputeEnvironments[0].Status) {
+	case string(svcapitypes.CEStatus_CREATING):
+		fmt.Println("postObserve: CEStatus_CREATING ")
+		cr.SetConditions(xpv1.Creating())
+	case string(svcapitypes.CEStatus_DELETING):
+		fmt.Println("postObserve: CEStatus_DELETING ")
+		cr.SetConditions(xpv1.Deleting())
+	case string(svcapitypes.CEStatus_VALID):
+		fmt.Println("postObserve: CEStatus_VALID ")
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.CEStatus_INVALID):
+		fmt.Println("postObserve: CEStatus_INVALID ")
+		cr.SetConditions(xpv1.Unavailable().WithMessage(*resp.ComputeEnvironments[0].StatusReason))
```

Comment:
Plain deref is unsafe and might lead to a panic. Better use helper for that:

```suggestion
	cr.SetConditions(xpv1.Unavailable().WithMessage(awsclients.StringValue(resp.ComputeEnvironments[0].StatusReason)))
```